components/cdc/src/delegate.rs

```diff
@@ -550,8 +551,10 @@ impl Delegate {
                     row_size = 0;
                 }
             }
-            // if the `txn_source` is not 0 and we should filter it out, skip this event.
-            if row.txn_source != 0 && filter_loop {
+            let lossy_ddl_filter = TxnSource::is_lossy_ddl_reorg_source_set(row.txn_source);
+            let cdc_write_filter =
+                TxnSource::is_cdc_write_source_set(row.txn_source) && filter_loop;
+            if lossy_ddl_filter || cdc_write_filter {
```

Comment:
It's better to take shortcut for all conditions here.

Reply:
Sorry, I don't get what you mean. Can you explain a bit?